src/platform/webengine/BlinkWebViewProfileHelper.h

```diff
@@ -40,14 +43,22 @@ const char kWebSQL[] = "webSQL";
 
 class BlinkWebViewProfileHelper {
 public:
-    BlinkWebViewProfileHelper() {}
-    virtual ~BlinkWebViewProfileHelper() {}
+    static BlinkWebViewProfileHelper* instance();
 
     static void clearBrowsingData(const int removeBrowsingDataMask,
         webos::WebViewProfile* profile = nullptr);
     static void clearDefaultBrowsingData(const int removeBrowsingDataMask);
     static int maskForBrowsingDataType(const char* key);
-    static void setProxyRules(const std::string& proxy_rules);
+    webos::WebViewProfile* getProfile(const std::string& app_id);
+    void buildProfile(const std::string& app_id, const std::string& proxy_rules);
+
+private:
+    BlinkWebViewProfileHelper() {}
+    virtual ~BlinkWebViewProfileHelper() = default;
+
+    std::map<const std::string, webos::WebViewProfile*> app_profile_map_;
+
+    DISALLOW_COPY_AND_ASSIGN(BlinkWebViewProfileHelper);
```

Comment:
You need to replace this with the declarations used in Chromium for it:
```
BlinkWebViewProfileHelper(const BlinkWebViewProfileHelper&) = delete;
BlinkWebViewProfileHelper& operator=(const BlinkWebViewProfileHelper&) = delete;

```

Reply:
I saw there's no other usage of this feature. Should I go for it anyway?